File: julia_scripts/new_seirp/weights_no_trb_pos_char.jl
using Oscar
R, vars = PolynomialRing(FiniteField(11863279), "s_8, i_8, s_7, i_7, e_7, b_7, s_6, i_6, e_6, b_6, s_5, i_5, e_5, b_5, s_4, i_4, e_4, b_4, s_3, i_3, e_3, b_3, s_2, i_2, e_2, b_2, s_1, i_1, e_1, b_1, s_0, r_0, p_0, i_0, e_0, b_0, z_aux, w_aux, a_e, a_i, k, mu, rho")
I = ideal(R, [1216027741153928-i_0-s_0, -e_0^2*k+b_0*i_0+i_0*mu+i_1, a_e*e_0^2*s_0+a_i*i_0*s_0+s_1, 372689350287209-b_0, b_1, -i_1-s_1-468937146127136725936897735554460858294147442, i_0*b_1-e_1^2*k+(b_0+mu)*i_1+i_2, a_e*s_0*e_1^2+a_i*s_0*i_1+(a_e*e_0^2+a_i*i_0)*s_1+s_2, e_0^2*rho^3-a_e*e_0^2*s_0-a_i*i_0*s_0+e_0^2*k+e_1^2, -i_2-s_2+199524261855035525671828780529078160071166027302104351687153054778297003390, 2*i_1*b_1+i_0*b_2-e_2^2*k+(b_0+mu)*i_2+i_3, (e_0^2*s_2+2*e_1^2*s_1+e_2^2*s_0)*a_e+(i_0*s_2+2*i_1*s_1+i_2*s_0)*a_i+s_3, b_2, (rho^3-a_e*s_0+k)*e_1^2-a_e*e_0^2*s_1+(-i_0*s_1-i_1*s_0)*a_i+e_2^2, -i_3-s_3+116235765894175674787045752638896226592777029699825163142503160752951015376446923422305002832419622436546, 3*i_2*b_1+3*i_1*b_2+i_0*b_3-e_3^2*k+(b_0+mu)*i_3+i_4, (e_0^2*s_3+3*e_1^2*s_2+3*e_2^2*s_1+e_3^2*s_0)*a_e+(i_0*s_3+3*i_1*s_2+3*i_2*s_1+i_3*s_0)*a_i+s_4, b_3, (-e_0^2*s_2-2*e_1^2*s_1-e_2^2*s_0)*a_e+(-i_0*s_2-2*i_1*s_1-i_2*s_0)*a_i+(rho^3+k)*e_2^2+e_3^2, -i_4-s_4-306187378281597790248384408029122489322540469407339531805850021745119973516914328472557686907199791395897074790415198552488285649397778, 4*i_3*b_1+6*i_2*b_2+4*i_1*b_3+i_0*b_4-e_4^2*k+(b_0+mu)*i_4+i_5, (e_0^2*s_4+4*e_1^2*s_3+6*e_2^2*s_2+4*e_3^2*s_1+e_4^2*s_0)*a_e+(i_0*s_4+4*i_1*s_3+6*i_2*s_2+4*i_3*s_1+i_4*s_0)*a_i+s_5, b_4, (-e_0^2*s_3-3*e_1^2*s_2-3*e_2^2*s_1-e_3^2*s_0)*a_e+(-i_0*s_3-3*i_1*s_2-3*i_2*s_1-i_3*s_0)*a_i+(rho^3+k)*e_3^2+e_4^2, -i_5-s_5+40095002841633521416051337774836256735700413135914857062810197157673534563501123892392310511153521938431148058330103255407290934915884043870266206233244810577601298, 5*i_4*b_1+10*i_3*b_2+10*i_2*b_3+5*i_1*b_4+i_0*b_5-e_5^2*k+(b_0+mu)*i_5+i_6, (e_0^2*s_5+5*e_1^2*s_4+10*e_2^2*s_3+10*e_3^2*s_2+5*e_4^2*s_1+e_5^2*s_0)*a_e+(i_0*s_5+5*i_1*s_4+10*i_2*s_3+10*i_3*s_2+5*i_4*s_1+i_5*s_0)*a_i+s_6, b_5, (-e_0^2*s_4-4*e_1^2*s_3-6*e_2^2*s_2-4*e_3^2*s_1-e_4^2*s_0)*a_e+(-i_0*s_4-4*i_1*s_3-6*i_2*s_2-4*i_3*s_1-i_4*s_0)*a_i+(rho^3+k)*e_4^2+e_5^2, -i_6-s_6+851687991352968111227913159365022678656374411202672898931100234546115474229017082689459431772569249239770131020172730540905698831034235047732133389161366678099669231519557567796054210551220469150, 6*i_5*b_1+15*i_4*b_2+20*i_3*b_3+15*i_2*b_4+6*i_1*b_5+i_0*b_6-e_6^2*k+(b_0+mu)*i_6+i_7, (e_0^2*s_6+6*e_1^2*s_5+15*e_2^2*s_4+20*e_3^2*s_3+15*e_4^2*s_2+6*e_5^2*s_1+e_6^2*s_0)*a_e+(i_0*s_6+6*i_1*s_5+15*i_2*s_4+20*i_3*s_3+15*i_4*s_2+6*i_5*s_1+i_6*s_0)*a_i+s_7, b_6, (-e_0^2*s_5-5*e_1^2*s_4-10*e_2^2*s_3-10*e_3^2*s_2-5*e_4^2*s_1-e_5^2*s_0)*a_e+(-i_0*s_5-5*i_1*s_4-10*i_2*s_3-10*i_3*s_2-5*i_4*s_1-i_5*s_0)*a_i+(rho^3+k)*e_5^2+e_6^2, -i_7-s_7-1180134802312987710906473782203358497428388105768150038340065657609875615332944753631312293312012091087948736423006818965289882935784627358895337829947891947405157557047816990941794443016651463249170852742681603322714677468542, 7*i_6*b_1+21*i_5*b_2+35*i_4*b_3+35*i_3*b_4+21*i_2*b_5+7*i_1*b_6+i_0*b_7-e_7^2*k+(b_0+mu)*i_7+i_8, (e_0^2*s_7+7*e_1^2*s_6+21*e_2^2*s_5+35*e_3^2*s_4+35*e_4^2*s_3+21*e_5^2*s_2+7*e_6^2*s_1+e_7^2*s_0)*a_e+(i_0*s_7+7*i_1*s_6+21*i_2*s_5+35*i_3*s_4+35*i_4*s_3+21*i_5*s_2+7*i_6*s_1+i_7*s_0)*a_i+s_8, b_7, (-e_0^2*s_6-6*e_1^2*s_5-15*e_2^2*s_4-20*e_3^2*s_3-15*e_4^2*s_2-6*e_5^2*s_1-e_6^2*s_0)*a_e+(-i_0*s_6-6*i_1*s_5-15*i_2*s_4-20*i_3*s_3-15*i_4*s_2-6*i_5*s_1-i_6*s_0)*a_i+(rho^3+k)*e_6^2+e_7^2, -i_8-s_8-3040583250952869529910019965457235634890047509998408300799411749605050307823026472432672145328226428256452928880873628219933524388036905531905469812307416750005610319583470186222156045495918750341941717359679738885273765131395776005668746711432475327856626, -b_1, -b_2, -b_3, -b_4, -b_5, -b_6, -b_7, 772347742513698-p_0, 988248791655614-r_0, z_aux^2-1])
gb = f4(I)
# {b_0 = b_0, b_1 = b_1, b_2 = b_2, b_3 = b_3, b_4 = b_4, b_5 = b_5, b_6 = b_6, b_7 = b_7, e_0 = e_0^2, e_1 = e_1^2, e_2 = e_2^2, e_3 = e_3^2, e_4 = e_4^2, e_5 = e_5^2, e_6 = e_6^2, e_7 = e_7^2, i_0 = i_0, i_1 = i_1, i_2 = i_2, i_3 = i_3, i_4 = i_4, i_5 = i_5, i_6 = i_6, i_7 = i_7, i_8 = i_8, rho = rho^3, s_0 = s_0, s_1 = s_1, s_2 = s_2, s_3 = s_3, s_4 = s_4, s_5 = s_5, s_6 = s_6, s_7 = s_7, s_8 = s_8, z_aux = z_aux^2}

File: magma_scripts/new_seir2/weights_no_trb_pos_char.m
SetNthreads(64);
Q := GF(11863279);
SetVerbose("Faugere", 2);
P<r_9, i_9, r_8, i_8, e_8, s_7, r_7, i_7, e_7, s_6, r_6, i_6, e_6, s_5, r_5, i_5, e_5, s_4, r_4, i_4, e_4, s_3, r_3, i_3, e_3, s_2, r_2, i_2, e_2, s_1, r_1, i_1, e_1, s_0, r_0, i_0, e_0, z_aux, w_aux, b, d0, eps, mu, rho>:= PolynomialRing(Q, 44, "grevlex");
G := ideal< P | 668740727985434478-i_0-r_0, i_0*rho^3-e_0^2*eps+i_0*mu+i_1, -i_0*rho^3+d0*r_0+r_1, -i_1-r_1+784612527403574732855439845642966487, -eps*e_1^2+(rho^3+mu)*i_1+i_2, -i_1*rho^3+d0*r_1+r_2, -b^3*i_0*s_0^3+e_0^2*eps+e_1^2, -i_2-r_2+2487778755737818441698233709124232101580517904922559707897866333295556, -eps*e_2^2+(rho^3+mu)*i_2+i_3, -i_2*rho^3+d0*r_2+r_3, -b^3*i_0*s_1^3-b^3*i_1*s_0^3+e_1^2*eps+e_2^2, b^3*i_0*s_0^3+s_1^3, -i_3-r_3-73160296065823370099332128637084003813238097949116019706247179221135706261284636738459579145297447073363, -eps*e_3^2+(rho^3+mu)*i_3+i_4, -i_3*rho^3+d0*r_3+r_4, (-i_0*s_2^3-2*i_1*s_1^3-i_2*s_0^3)*b^3+eps*e_2^2+e_3^2, b^3*i_0*s_1^3+b^3*i_1*s_0^3+s_2^3, -i_4-r_4+2151489117789942799644091746969014341662944108012071442977365066985271453430823156927057370343837389128148177151930506878284786567644703114, -eps*e_4^2+(rho^3+mu)*i_4+i_5, -i_4*rho^3+d0*r_4+r_5, (-i_0*s_3^3-3*i_1*s_2^3-3*i_2*s_1^3-i_3*s_0^3)*b^3+eps*e_3^2+e_4^2, (i_0*s_2^3+2*i_1*s_1^3+i_2*s_0^3)*b^3+s_3^3, -i_5-r_5-63270731160024947503815858180687168054710453933404455528564147363555493430479317785321703816550874231460253938225137000346169676604605653042173684554143140531286838512283397, -eps*e_5^2+(rho^3+mu)*i_5+i_6, -i_5*rho^3+d0*r_5+r_6, (-i_0*s_4^3-4*i_1*s_3^3-6*i_2*s_2^3-4*i_3*s_1^3-i_4*s_0^3)*b^3+eps*e_4^2+e_5^2, (i_0*s_3^3+3*i_1*s_2^3+3*i_2*s_1^3+i_3*s_0^3)*b^3+s_4^3, -i_6-r_6+1860657991910416728971054988790553982465647687689094631618814562711304131221797291044914816212087597599685771903949094940314723026716546203598821755599348553445130802220014878618983980105494591364901657872680, -eps*e_6^2+(rho^3+mu)*i_6+i_7, -i_6*rho^3+d0*r_6+r_7, (-i_0*s_5^3-5*i_1*s_4^3-10*i_2*s_3^3-10*i_3*s_2^3-5*i_4*s_1^3-i_5*s_0^3)*b^3+eps*e_5^2+e_6^2, (i_0*s_4^3+4*i_1*s_3^3+6*i_2*s_2^3+4*i_3*s_1^3+i_4*s_0^3)*b^3+s_5^3, -i_7-r_7-54718004666389248059536473893184613196788403098140136756971211409383586667242013178231646337294754810160297237143654087310437591221906408184754845203839588749503400694101248266698046770025514410466350546635724814022423618232741563253260829943, -eps*e_7^2+(rho^3+mu)*i_7+i_8, -i_7*rho^3+d0*r_7+r_8, (-i_0*s_6^3-6*i_1*s_5^3-15*i_2*s_4^3-20*i_3*s_3^3-15*i_4*s_2^3-6*i_5*s_1^3-i_6*s_0^3)*b^3+eps*e_6^2+e_7^2, (i_0*s_5^3+5*i_1*s_4^3+10*i_2*s_3^3+10*i_3*s_2^3+5*i_4*s_1^3+i_5*s_0^3)*b^3+s_6^3, -i_8-r_8+1609140447996502371962207024402034257500038250380414832163796375704696703412022833682797479786417993788161557486303581738045821922552895883300131073607586551133412461996654110757374118969916388370923344810171320917748814722568458557405212491464182950671744984865894284237137662, -eps*e_8^2+(rho^3+mu)*i_8+i_9, -i_8*rho^3+d0*r_8+r_9, (-i_0*s_7^3-7*i_1*s_6^3-21*i_2*s_5^3-35*i_3*s_4^3-35*i_4*s_3^3-21*i_5*s_2^3-7*i_6*s_1^3-i_7*s_0^3)*b^3+eps*e_7^2+e_8^2, (i_0*s_6^3+6*i_1*s_5^3+15*i_2*s_4^3+20*i_3*s_3^3+15*i_4*s_2^3+6*i_5*s_1^3+i_6*s_0^3)*b^3+s_7^3, -i_9-r_9-47321407225379050236774078325943717714853779219574988544512724713844968205475315289374580068698866715952639756061480154166196098999787778067671324237292596147100763691003578213859949987128688021530550265705650211100304199058765493217516296477510029562010023137962391700740464101577707382133051322141786307175937, z_aux^2-1>;
time GroebnerBasis(G);// {b = b^3, e_0 = e_0^2, e_1 = e_1^2, e_2 = e_2^2, e_3 = e_3^2, e_4 = e_4^2, e_5 = e_5^2, e_6 = e_6^2, e_7 = e_7^2, e_8 = e_8^2, i_0 = i_0, i_1 = i_1, i_2 = i_2, i_3 = i_3, i_4 = i_4, i_5 = i_5, i_6 = i_6, i_7 = i_7, i_8 = i_8, i_9 = i_9, r_0 = r_0, r_1 = r_1, r_2 = r_2, r_3 = r_3, r_4 = r_4, r_5 = r_5, r_6 = r_6, r_7 = r_7, r_8 = r_8, r_9 = r_9, rho = rho^3, s_0 = s_0^3, s_1 = s_1^3, s_2 = s_2^3, s_3 = s_3^3, s_4 = s_4^3, s_5 = s_5^3, s_6 = s_6^3, s_7 = s_7^3, z_aux = z_aux^2}
quit;